mongo/errors.go

```diff
@@ -71,6 +73,60 @@ func replaceErrors(err error) error {
 	return err
 }
 
+// IsDuplicateKeyError returns true if err is a duplicate key error
+func IsDuplicateKeyError(err error) bool {
+	// handles SERVER-7164 and SERVER-11493
+	for ; err != nil; err = unwrap(err) {
+		if e, ok := err.(ServerError); ok {
+			return e.HasErrorCode(11000) || e.HasErrorCode(11001) || e.HasErrorCode(12582) ||
```

Comment:
Do we keep error codes as constants anywhere?

Reply:
Gotcha, wonder if it's worth making a `duplicateKeyCodes []int32` var for these three as in `drivers/errors`. Up to you, though.